[Bug]: wrong region used for requests to route53

### Terraform Core Version

1.5.0

### AWS Provider Version

5.21.0

### Affected Resource(s)

aws_route53_record

### Expected Behavior

I should be able to use the fips endpoint: https://route53-fips.amazonaws.com. However, it seems the provider hard-codes the use of the us-west-2 region when making requests to route53 and that isn't a valid region for that endpoint.

### Actual Behavior

If i use a provider configuration that uses the `us-east-1` region, and specify 
```terraform
endpoints {
  route53 = "https://route53-fips.amazonaws.com"
}
```

then when I try to refresh the state (including as part of a plan) I get an error that the " SignatureDoesNotMatch: Credential should be scoped to a valid region"

### Relevant Error/Panic Output Snippet

```shell
╷
│ Error: reading Route 53 Record (<REDACTED>): SignatureDoesNotMatch: Credential should be scoped to a valid region.
│ 	status code: 403, request id: c0729919-504b-4889-9b9a-ba49181f09fc
│ 
│   with aws_route53_record.example,
│   on ses.tf line 9, in resource "aws_route53_record" "example":
│    9: resource "aws_route53_record" "example" {
│ 
╵
```


### Terraform Configuration Files

```terraform
provider "aws" {
  alias = "example"
  # include AWS credentials source
  region = "us-east-1"

  endpoints {
    route53 = "https://route53-fips.amazonaws.com"
  }
}

resource "aws_route53_record" "example" {
  provider = aws.example
  zone_id  = var.zone_id
  name     = ""
  type     = "TXT"
  ttl      = 60
  records  = ["example"]
}
```

### Steps to Reproduce

Create the resource without the custom route53 endpoint. Then run `terraform apply -refresh-only` to attempt to refresh the state.

### Debug Output

```
2023-10-19T12:54:43.808-0600 [DEBUG] provider.terraform-provider-aws_v5.21.0_x5: HTTP Response Received: rpc.service="Route 53" tf_mux_provider=*schema.GRPCProviderServer tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=dda458c2-5606-1bb9-dc6f-4fc926b1e7b3 http.response.header.content_type=text/xml @module=aws http.duration=427 http.response.header.x_amzn_requestid=8617f94b-5f9e-44c8-9d50-98ce16a38644 http.status_code=403 rpc.system=aws-api tf_resource_type=aws_route53_record aws.region=us-west-2 http.response.header.date="Thu, 19 Oct 2023 18:54:42 GMT" rpc.method=GetHostedZone tf_rpc=ReadResource @caller=github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.38/logger.go:157 http.response.body="<?xml version="1.0"?>
<ErrorResponse xmlns="https://route53.amazonaws.com/doc/2013-04-01/"><Error><Type>Sender</Type><Code>SignatureDoesNotMatch</Code><Message>Credential should be scoped to a valid region. </Message></Error><RequestId>8617f94b-5f9e-44c8-9d50-98ce16a38644</RequestId></ErrorResponse>
" http.response_content_length=300 tf_aws.sdk=aws-sdk-go timestamp=2023-10-19T12:54:43.808-0600
2023-10-19T12:54:43.809-0600 [ERROR] provider.terraform-provider-aws_v5.21.0_x5: Response contains error diagnostic: tf_rpc=ReadResource @module=sdk.proto diagnostic_severity=ERROR diagnostic_summary="reading Route 53 Record (<REDACTED>__TXT): SignatureDoesNotMatch: Credential should be scoped to a valid region.
	status code: 403, request id: 8617f94b-5f9e-44c8-9d50-98ce16a38644" tf_proto_version=5.4 tf_resource_type=aws_route53_record @caller=github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/diag/diagnostics.go:58 diagnostic_detail= tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=dda458c2-5606-1bb9-dc6f-4fc926b1e7b3 timestamp=2023-10-19T12:54:43.809-0600
2023-10-19T12:54:43.809-0600 [ERROR] vertex "aws_route53_record.example" error: reading Route 53 Record (<REDACTED>__TXT): SignatureDoesNotMatch: Credential should be scoped to a valid region.
	status code: 403, request id: 8617f94b-5f9e-44c8-9d50-98ce16a38644
2023-10-19T12:54:43.809-0600 [ERROR] vertex "aws_route53_record.example (expand)" error: reading Route 53 Record (<REDACTED>__TXT): SignatureDoesNotMatch: Credential should be scoped to a valid region.
```


Notice that the request uses the "us-west-2" region, despite the fact that I have specified the "us-east-1" region.

### Panic Output

_No response_

### Important Factoids

_No response_

### References

* https://github.com/hashicorp/terraform-provider-aws/issues/33952 (which is why I am specifying the endpoint instead of using `use_fips_endpoints`)
* https://aws.amazon.com/compliance/fips/#FIPS_Endpoints_by_Service


### Would you like to implement a fix?

None

--- a/.changelog/37565.txt
+++ b/.changelog/37565.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_route53_record: Change region default to us-east-1
+```

--- a/internal/service/route53/service_package.go
+++ b/internal/service/route53/service_package.go
@@ -21,9 +21,8 @@ func (p *servicePackage) NewConn(ctx context.Context, m map[string]any) (*route5
 	// Force "global" services to correct Regions.
 	switch m["partition"].(string) {
 	case endpoints_sdkv1.AwsPartitionID:
-		if aws_sdkv1.StringValue(config.Endpoint) == "" {
-			config.Region = aws_sdkv1.String(endpoints_sdkv1.UsWest2RegionID)
-		}
+		// https://docs.aws.amazon.com/general/latest/gr/r53.html Setting default to us-east-1
+		config.Region = aws_sdkv1.String(endpoints_sdkv1.UsEast1RegionID)
 	case endpoints_sdkv1.AwsCnPartitionID:
 		// The AWS Go SDK is missing endpoint information for Route 53 in the AWS China partition.
 		// This can likely be removed in the future.

--- a/internal/service/route53/service_package.go
+++ b/internal/service/route53/service_package.go
@@ -21,7 +21,9 @@ func (p *servicePackage) NewConn(ctx context.Context, m map[string]any) (*route5
 	// Force "global" services to correct Regions.
 	switch m["partition"].(string) {
 	case endpoints_sdkv1.AwsPartitionID:
-		config.Region = aws_sdkv1.String(endpoints_sdkv1.UsWest2RegionID)
+		if aws_sdkv1.StringValue(config.Endpoint) == "" {
+			config.Region = aws_sdkv1.String(endpoints_sdkv1.UsWest2RegionID)
+		}
 	case endpoints_sdkv1.AwsCnPartitionID:
 		// The AWS Go SDK is missing endpoint information for Route 53 in the AWS China partition.
 		// This can likely be removed in the future.

